pkg/auth/authproviders/oidc/backend_impl.go

```diff
@@ -185,25 +186,27 @@ func (p *backendImpl) verifyIDToken(ctx context.Context, rawIDToken string, nonc
 		return nil, err
 	}
 
+	// Special case: in the case of an ID token being presented which has an empty group attribute, we attempt to
+	// enrich data from the userinfo endpoint. The rationale behind this is that some OIDC providers we've seen
+	// (i.e. GitLab) only present the complete group membership in the userinfo endpoint, not within the ID token.
+	if hasEmptyGroups(externalClaims) && rawAccessToken != "" {
+		externalClaims, err = p.fetchUserInfoClaims(ctx, rawAccessToken)
```

Comment:
I haven't found a concrete RFC for this, but the conventional knowledge about this is that the ID token may have additional claims, but in it's raw form is to signal the client that the user is authenticated. The user info endpoint may then provide additional information about the user, depending on the scope. I'd say it's safe to assume that the claims presented by the user info endpoint are at least equal to the claims presented by the ID token.

We can also just take the `groups` claim values here instead of overwriting everything, this seems like the most sanest solution then.

Reply:
I think I'm fine with whatever for now (because I don't know better :P). Just leave a consistent note.